lang/python314-devel/Portfile

```diff
@@ -59,6 +64,9 @@ depends_run         port:python_select-${pythonVerNoDot} \
                     port:python3_select-${pythonVerNoDot}
 
 compiler.c_standard 2011
+# libintvector.h:230:9: error: unknown type name '__m256i' on macOS < 10.11
+compiler.blacklist-append \
+                    {clang < 800}
```

Comment:
What changed to introduce this requirement?

Reply:
I suspect it has to do with the fact that older clang versions did not enable avx by default, and needed an extra flag sent. This was fixed in approx llvm clang-7.

these were added here:

https://github.com/python/cpython/commit/325e9b8ef400b86fb077aa40d5cb8cec6e4df7bb

see this:

https://github.com/llvm/llvm-project/blob/9a95c097d0466c594f40a4ba9ced8a155574fdff/llvm/lib/Support/BLAKE3/CMakeLists.txt#L53

and this:

https://github.com/BLAKE3-team/BLAKE3/issues/79

Reply:
That's right.

The updated HACL module actively utilizes intrinsic functions. They are enabled via `HACL_CAN_COMPILE_SIMD128` and `HACL_CAN_COMPILE_SIMD256` macros ([blocked on ARM64](https://github.com/python/cpython/blob/401bfc69d1258fde07141cd1a12af00aa44a8e0f/Modules/blake2module.c#L44) for obvious reasons).
I didn't block them because of the possible performance impact and since the software is still in alpha development.

Reply:
Rather than restrict compilers to this degree, perhaps it would be useful to consider adding the needed flag to enable avx512 like llvm did? That is probably also upstreamable, I would think.

Likely still need to restrict the compiler selection, but perhaps not as severely.

Reply:
Assuming that works, Josh could probably take that added avx512 flag upstream if he wanted to, using the LLVM installation as backup.

Reply:
A configure check for whether the compiler supports/needs `-mavx512vl` would be a more upstreamable fix.

Reply:
I deleted that part of the code and rebased the branch.

3.14 alpha 4 release is [expected today](https://peps.python.org/pep-0745/), but I want to see how these changes from this PR will affect buildbots.